src/app/client/src/app/modules/shared/services/navigate/router-navigation.service.spec.ts

```diff
@@ -1,17 +1,50 @@
 import { TestBed, inject } from '@angular/core/testing';
-import { Router } from '@angular/router';
+import { Router, NavigationEnd } from '@angular/router';
 import { configureTestSuite } from '@sunbird/test-util';
 import { RouterNavigationService } from './router-navigation.service';
+import { Observable, of } from 'rxjs';
 
-xdescribe('RouterNavigationService', () => {
+class RouterStub {
+  public navigationEnd = new NavigationEnd(0, '/explore', '/explore');
+  public url = '';
+  public events = new Observable(observer => {
+    observer.next(this.navigationEnd);
+    observer.complete();
+  });
+  public navigate() {
+    return of({});
+  }
+}
+
+describe('RouterNavigationService', () => {
   configureTestSuite();
   beforeEach(() => {
     TestBed.configureTestingModule({
-      providers: [RouterNavigationService, Router]
+      providers: [
+        RouterNavigationService,
+        { provide: Router, useClass: RouterStub }
+      ]
     });
   });
 
   it('should be created', inject([RouterNavigationService], (service: RouterNavigationService) => {
     expect(service).toBeTruthy();
   }));
+
+  it('should call navigateToParentUrl', inject([RouterNavigationService], (service: RouterNavigationService) => {
+    let routeMock: any = {
+      snapshot: {},
+      parent: {
+        url: [
+          {
+            path: '/home'
+          }
+        ]
+      }
+    };
+    spyOn(service, 'navigateToParentUrl').and.returnValue(of({}));
+    service.navigateToParentUrl(routeMock);
+    expect(service.navigateToParentUrl).toHaveBeenCalled();
```

Comment:
Actual assertion should be you have to check for "component.parentUrl" having the right value what you are expecting.

Reply:
Changes have been updated.